MinecraftClient/Mapping/Movement.cs

```diff
@@ -129,62 +129,89 @@ public static Queue<Location> Move2Steps(Location start, Location goal, ref doub
-                    { //reconstruct_path(Came_From, goal)
-                        List<Location> total_path = new List<Location>(new[] { current });
-                        while (Came_From.ContainsKey(current))
-                        {
-                            current = Came_From[current];
-                            total_path.Add(current);
-                        }
-                        total_path.Reverse();
-                        result = new Queue<Location>(total_path);
-                    }
-                    OpenSet.Remove(current);
-                    ClosedSet.Add(current);
-                    foreach (Location neighbor in GetAvailableMoves(world, current, allowUnsafe))
-                    {
-                        if (ClosedSet.Contains(neighbor))
-                            continue;		// Ignore the neighbor which is already evaluated.
-                        int tentative_g_score = g_score[current] + (int)current.DistanceSquared(neighbor); //dist_between(current,neighbor) // length of this path.
-                        if (!OpenSet.Contains(neighbor))	// Discover a new node
-                            OpenSet.Add(neighbor);
-                        else if (tentative_g_score >= g_score[neighbor])
-                            continue;		// This is not a better path.
+                    if (ct.IsCancellationRequested)
+                        break;
+                    if (ClosedSet.Contains(neighbor))
+                        continue;       // Ignore the neighbor which is already evaluated.
+                    int tentative_g_score = g_score[current] + (int)current.DistanceSquared(neighbor); //dist_between(current,neighbor) // length of this path.
+                    if (!OpenSet.Contains(neighbor))    // Discover a new node
+                        OpenSet.Add(neighbor);
+                    else if (tentative_g_score >= g_score[neighbor])
+                        continue;       // This is not a better path.
 
-                        // This path is the best until now. Record it!
-                        Came_From[neighbor] = current;
-                        g_score[neighbor] = tentative_g_score;
-                        f_score[neighbor] = g_score[neighbor] + (int)neighbor.DistanceSquared(goal); //heuristic_cost_estimate(neighbor, goal)
-                    }
+                    // This path is the best until now. Record it!
+                    Came_From[neighbor] = current;
+                    g_score[neighbor] = tentative_g_score;
+                    f_score[neighbor] = g_score[neighbor] + (int)neighbor.DistanceSquared(goal); //heuristic_cost_estimate(neighbor, goal)
                 }
-            }, TimeSpan.FromSeconds(5));
+            }
 
-            return result;
+            // Could not be reached. Set the path to the closest possible location
+            if (goal.DistanceSquared(closestGoal) <= maxOffset || maxOffset == int.MaxValue)            
```

Comment:
Skip calculating distance if offset is already `int.MaxValue`
```suggestion
            // Goal could not be reached. Set the path to the closest location if close enough
            if (maxOffset == int.MaxValue || goal.DistanceSquared(closestGoal) <= maxOffset)            
```